An important aspect of Statistics and Probability is the ability to count large numbers. To count large numbers we need to learn about Permutations and Combinations.

### Permuatation

Suppose you have to arrange five objects into a sequence. How many ways can you
do that?

Here the order of the objects matters, hence we will use permuation to count the number of ways we can arrange the objects.

In [1]:
5*4*3*2*1

[1] 120

In [2]:
factorial(5) / factorial(5-5)

[1] 120

Permuation Examples.

A Horse race has 12 horses. How many ways can we get the 1st, 2nd and 3rd place finish?

In [3]:
factorial(12) / factorial(12 - 3)

[1] 1320

There 1320 different ways we can arrange the top 3 winners.

During a baseball season, an MLB team has 25 players on a roster and a manager can only select 9. How many different lineups(batting order) can a manager make use of?

We have 25 choose 9.

In [4]:
factorial(25) / factorial(25 - 9)

[1] 741354768000

We have more than 7.41 Billion possibilities of lineups.

A website requires you to make a 16 digit passwords. You are allowed to use only the letters(upper and lowercase) and numbers and special characters. How many different 16 digit passwords can you make?

In [5]:
# Options
upper = 26
lower = 26
numbers = 10
special = 13

# Total
total = upper + lower + numbers + special


choose = 16

perm = factorial(total) / factorial(total - choose)

print(perm)

[1] 1.788908e+29


That's a very large number, and the password may be very hard to crack. Its a good idea to have a strong password.

### Combinations

In combinations, the order doesnt matter. Lets go back to the base ball example.

During a baseball season, an MLB team has 25 players on a roster and a manager can only select 9. How many different groups/teams of players can a manager choose?

In [6]:
factorial(25) /(factorial(9) * factorial(25 - 9))

[1] 2042975

Now when the order does not matter we have a little more than 2 million ways to choose our team. Thats much less than the 7.41 Billion we had before.

There are 30 stocks in the Dow Jones. Each month we decide to invest in 1 stock for the next 6 months. 

How many unique groups of 6 stock portfolio are possible?

In [7]:
factorial(30) /(factorial(6) * factorial(30 - 6))

[1] 593775

We can have 593775 unique portfolios of stocks.

----
More examples.

A pizza shop offers a special medium pizza with 3 different toppins for $9.99. There are 10 toppings in total. How many different pizza types can be made?

In [8]:
comb = factorial(10) /(factorial(3) * factorial(10 - 3))

cat("We can have", comb, "different types of pizzas")

We can have 120 different types of pizzas

A lottery requires you to get 6 numbers correct from a possible choice of 52 in any order. How many different groups of winning lottery numbers are there.

In [9]:
comb = factorial(52) /(factorial(6) * factorial(52 - 6))
cat("We can have", comb, "different possible ways.")

We can have 20358520 different possible ways.